src/include/common/serializer/writer.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <cstdint>
+
+namespace kuzu {
+namespace common {
+
+class Writer {
+public:
+    virtual void write(const uint8_t* data, uint64_t size) = 0;
+    virtual ~Writer(){};
```

Comment:
I think our convension is to use default for trivial deconstructors.
virtual ~Writer() = default;